src/transporters.ts

```diff
@@ -57,18 +57,19 @@ export class DefaultTransporter {
    * @return Configured options.
    */
   configure(opts: GaxiosOptions = {}): GaxiosOptions {
-    opts.headers = opts.headers || {};
+    const headers = Object.assign({}, opts.headers || {});
```

Comment:
For my learning - what's the value of doing the clone here as opposed to just modifying the headers in place?  As far as I can tell, we just assign it back to the options object at the end anyway.

Reply:
It's a bit defensive, but if someone had written code like this:

```
const headers = {'User-Agent': 'my-app'};
const options = {
  headers
}
auth.configure(options);
console.info(headers);
```

without the copy, we will delete the key on their original headers object, not just `options.headers`, because options.headers is populated as a reference.